### PR TITLE
sql: replace deprecated descs.Collection method calls

### PR DESCRIPTION
**Pull request by postamar**
descs: remove GetObjectNamesAndIDs method

This commit replaces usages of that method with calls to
GetAllObjectsInSchema.

Informs #64089.

Release note: None

----

sql: amortize schema descriptor lookups in vtable generation

This gets rid of some deprecated descs.Collection method calls.

Informs #64089.

Release note: None

----

scbuild,scdeps: replace deprecated descs.Collection method calls

Informs #64089.

Release note: None

### CODE REVIEW

**Comment by postamar** [comment]
@ajwerner this is the PR I alluded to in https://github.com/cockroachdb/cockroach/pull/93543 which undoes the rttanalysis regression.

File: pkg/sql/schema_resolver.go [hunk 1]
```diff
@@ -69,18 +70,27 @@ type schemaResolver struct {
 // GetObjectNamesAndIDs implements the resolver.SchemaResolver interface.
 func (sr *schemaResolver) GetObjectNamesAndIDs(
```
Comment (ajwerner) [inline pkg/sql/schema_resolver.go:71]: I feel like we've got a terminology problem these days: is a function an object? 

By my old definition, it is, but it isn't here.

**Review by ajwerner — APPROVED** [review]
I'm cool with this change. More generally I think functions have muddled our terminology in a bad way.

File: pkg/sql/schema_resolver.go (hunk 1 above)
Comment (postamar) [reply pkg/sql/schema_resolver.go:71]: It is, but it's also a special kind of object in that unlike other objects we don't expect functions to have a namespace entry.

I completely agree with you in general, btw. Untangling `GetObjectNamesAndIDs` is going to be its own unit of work focused on the resolver interface, and it's something I want to do this week.

**Comment by postamar** [comment]
Thanks for the review!

bors r+

**Pull request merged**